### ISSUE TITLE
New weapon: Order of Chaos

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#239]
Will you harness the chaos? Or will it consume you?

copy from Revolver
firerate halved.
cannot take ExtendedMags.

This being, effectively, a revolver, this gun has 6 "slots". Each of these slots can accept one module, which specifies how the gun will behave when firing that module.

Possible modules:
 - a Flintlock shot
 - a Pump-like shotgun shot
 - a sniper shot
 - a grenade launcher shot
 - an AOE explodes on top of you, and you dash forward
 - whatever else you can think of, lmao

Now, here's the key. The set of modules installed in this weapon is *random*, and is decided at the beginning of the round. So each round, this weapon will offer a relatively fresh and unique experience. Some rounds it will be a shotgun with some occasional grenades. Other times it will alternate sniper fire and dashes. Most rounds it will be a complete mess of everything, in a specific order.

**Comment by ghost** [comment]
I think a grenade launcher might be unbalanced to add, otherwise it seems fine for an rng weapon. I feel like it would be unpreffered by players, because of how random it is, and because they have the choice to simply choose a more consistent weapon, that might be more stronger.

**Comment by deferred-impact** [comment]
Well it's not very random, it is only randomized at the beginning of the match, and then for the entirety of that match, all players have the same configuration for this weapon

**Comment by ghost** [comment]
Yet, seeing the random choices at the start of the map, they can be just as bad or just as good for a player, without them knowing (I would assume the configuration is discreet to the player before they use the weapon), which would enforce them to consider not using it. Would you rather prefer to use any other weapon, that has more 'permanent' stats, or something that is completely random, has a slight chance of being useful in a fight, and has the potential to harm you?
Though I do think this idea is at a good start, it could be improvised, maybe with less rng.

**Comment by deferred-impact** [comment]
I mean, i get your point, yes. Other weapons are more reliable and in many scenarios, this weapon will roll a suboptimal configuration. But since you're not forced to use this weapon, i don't think that's a massive issue. You can always switch to a different one if you find this one to be bad in the current round

**Comment by ghost** [comment]
Yes, but I think that will lead to it being underused, and useless, which is why I want to improve it in some way, to make it more appealing, and more usable and preferable by other players.
